test/basic/ecs_compute_update/ecs_compute_update_test.go

```diff
@@ -184,6 +184,13 @@ func TestECSComputeUpdate(t *testing.T) {
 		)
 		assert.Nil(t, err)
 		assertCommonValues(t, vpcId, updatedECSComputeId, vpcAsset, vpcAws, updatedECSComputeAsset, updatedECSClusterAws, updatedECSServiceAws)
+		assert.Equal(t, len(ecsServiceAws.Deployments), 1)
+		ecsTaskDefinitionAws := terratest_aws.GetEcsTaskDefinition(t, "us-east-1", *ecsServiceAws.Deployments[0].TaskDefinition)
```

Comment:
How do we know that `Deployments[0]` is the latest deployment?

Reply:
isn't that entropic if it ever rolls back?

```
		assert.Equal(t, len(ecsServiceAws.Deployments), 1)
		ecsTaskDefinitionAws := terratest_aws.GetEcsTaskDefinition(t, "us-east-1", *ecsServiceAws.Deployments[0].TaskDefinition)
```

i ensure its length of 1 first, but i do think i can check event history (but that's on the deployment too) and won't really tell me this either

----

i think the best way for us to approach this test is maybe how we look at it in the UI. how do we know something changed? probably image tags or shas (or maybe task definition #, but that accumulates and gets increasingly difficult). i think for nginx on the updates, i can always switch between two versions and use those image tags to check if i'm in the expected one maybe?

Reply:
I mean regardless, there should probably be assertions for each value changing, so if it does roll back and there's a mismatch, it would catch it (i.e. no need to change the container for every change). But generally agree